Delete selected mocks and groups with keyboard

dmitrika opened the pull request:
Hello!

This pr is connected with #216.

It's my first contribution with code-changes, so feel free to point where I should correct it.
And I couldn't get how to cover it with test, any thoughts?

morsdyce commented on lib/ui/components/Mocks/Sidebar/MocksSidebar.js:
```diff
@@ -46,13 +47,29 @@ class MocksSidebar extends React.PureComponent {
     API.off(EVENTS.UPDATE_MOCK, this.reRender);
     API.off(EVENTS.UPDATE_GROUP, this.reRender);
+
+    document.removeEndEventListener("keydown", this.onKeyDown);
```
should be `removeEventListener` instead of `removeEndEventListener`

morsdyce commented on lib/ui/components/Mocks/Sidebar/MocksSidebar.js:
```diff
@@ -46,13 +47,29 @@ class MocksSidebar extends React.PureComponent {
 
+  onKeyDown = (event) => {
+    if (event.keyCode === KEYCODES.BACKSPACE || event.keyCode === KEYCODES.DELETE) {
+        const selectedGroupsIds = this.props.selectedGroups.map(group => group.id);
```
There is no need to do this logic yourself, we have State classes to handle these mutations so we can have them all in one place.

You can import `MocksState` which is the state for anything related directly to mocks (and groups) and use the `deleteSelected` method.

ilyagelman replied:
I guess we have to document all the states we have

ilyagelman commented:
@morsdyce Good question about focus... Obviously we would need undo for this.

What about adding a `sidebarInFocus` flag to the mocks state? It could be set to true when clicking inside the sidebar or on one of its items and unset with clickOutside or something.

morsdyce commented:
Good idea, I like it much better than assuming that if the mouse is in the sidebar area then you probably meant to delete.

I think this is the proper solution for now, undo will be important for this feature so we might want to release them together.

dmitrika commented:
Done with typo and MocksState.

I got idea with focus, all right.

And how particularly you see undo logic here? Is this a button or cmd + z hotkey? Or gmail like popover?

morsdyce commented:
We have a button which is hidden right now because the functionality isn't ready which might also be bound to a hotkey

ilyagelman commented:
@dmitrika I'd tackle the undo as a different issue, doesn't have to be done now

dmitrika commented:
I'm still want to finish the task, but waiting when I would be able to get DOM-node from styled-components and add 'click' event-listener to it.

Now it's kinda broken https://github.com/styled-components/styled-components/issues/629

dmitrika commented:
I added `sidebarInFocus` flag and toggle function to the MocksState.

Unfortunately, I couldn't get DOM-element from styled-components, despite bugfix, so I added cb to DnD component to get node from it.

morsdyce commented:
Thanks! I just went on vacation so will take me a while to review it.

On Apr 4, 2017 15:05, "Dmitry Rybin" <notifications@github.com> wrote:

> I added sidebarInFocus flag and toggle function to the MocksState.
>
> Unfortunately, I couldn't get DOM-element from styled-components, despite
> bugfix, so I added cb to DnD component to get node from it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/500tech/mimic/pull/222#issuecomment-291616242>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABlqItkGP0z4x5hTZWjM5ULnkqJbIyliks5rsqKSgaJpZM4Mhwwu>
> .
>

dmitrika commented:
@morsdyce Just want to ping you in case if you missed pr :)

morsdyce commented:
@dmitrika Don't worry, I didn't forget about you, I just came back from a month long vaction and have to deal with some work catch up stuff as well as preparing a few things to present mimic in oscon. I will check this as soon as I can which should be in the next few days.

Thanks for your help!

morsdyce commented:
@dmitrika I just checked it and it works beautifully! 
Thanks so much for the help :)

dmitrika commented:
Great, thanks!
